contracts/ColonyStorage.sol

```diff
@@ -116,6 +120,27 @@ contract ColonyStorage is CommonStorage, ColonyDataTypes, ColonyNetworkDataTypes
     _;
   }
 
+  modifier expenditureExists(uint256 _id) {
+    require(_id > 0 && _id <= expenditureCount, "colony-expenditure-does-not-exist");
+    _;
+  }
+
+  modifier expenditureActive(uint256 _id) {
+    require(expenditures[_id].status == ExpenditureStatus.Active, "colony-expenditure-not-active");
+    _;
+  }
+
+  modifier expenditureFinalized(uint256 _id) {
+    require(expenditures[_id].status != ExpenditureStatus.Cancelled, "colony-expenditure-cancelled");
+    require(expenditures[_id].status == ExpenditureStatus.Finalized, "colony-expenditure-not-finalized");
```

Comment:
It could be `Active`.

Reply:
If it were `Active`, `expenditures[_id].status == ExpenditureStatus.Finalized` would surely be false?

Reply:
I mean yes, but that's the point no? This modifier only let's finalized expenditures through, so if it's active the require will fail. I think of the first require as a courtesy since it makes the error message more helpful. I could have equivalently had `!= Active` but this seemed a bit better since it allow for more hypothetical states to be introduced in the future.

Reply:
>  I think of the first require as a courtesy since it makes the error message more helpful.

Yeah, this is why I don't like it. The modifier is `expenditureFinalized`, and there is a require that checks `expenditures[_id].status == ExpenditureStatus.Finalized`. An additional require is unnecessary - the gain going from `colony-expenditure-not-finalized` to `colony-expenditure-cancelled` is incredibly marginal, and increases the gas cost for every valid call.

Reply:
Ok, I can drop it.